R/backend-mssql.R

```diff
@@ -150,8 +150,31 @@
           sql_expr(DATEPART(QUARTER, !!x))
         }
       },
+    )
 
-    ),
+  if (!is(con, "DBIObject") || sub("\\..*", "", DBI::dbGetInfo(con)$db.version) >= 11) {
```

Comment:
Can you please pull this out into separate function? I think that will make the main thrust of the code easier to understand.

Reply:
Thanks Hadley. Just to clarify, would you like me to place everything inside the curly braces { *re-assignment of mssql_scalr* } as a function? Or even the 'test condition' (*!is.con(...) and *version >= 11*) should also be in a separate function?

Reply:
The test of whether or not an connection has an adequate version of SQL server

Reply:
okay, non-exported function `mssql_version()` returns a list of `ProductVersion_Major`/`ProductVersion_Minor`/`ProductVersion_Revision` (returns list of NULLs if argument `con` is not a valid `DBIObject`). `mssql_version()` is used to determine whether the MSSQL server version is >=11, in which case `sql_translate_env.Microsoft SQL Server` uses `TRY_CAST` instead of `CAST` where possible.

Also added to `tests/testthat/test-backend-mssql.R` to test the small non-exported function `mssql_interpret_version()` which is used to interpret the `$db.version` string returned by `DBI::dbGetInfo()`.